### PR TITLE
Add recipe for sharing state between Astro components

### LINKED ISSUE #3327: Missing docs: How to use `nanostores` with Astro components
Our [Share State Between Islands](https://docs.astro.build/en/core-concepts/sharing-state) guide explains how to use `nanostores` with a variety of UI frameworks. However, not everyone wants to include a UI framework, and might prefer to do it in pure JavaScript. Since a `script` in an Astro file isn't an island it makes sense that it's not documented here, but I feel it does deserve a place somewhere. The way I see it, there are a couple of ways to solve this:

1. "Rebrand" the guide as "Share State Between Components"
2. Create a new guide elsewhere specifically for Astro (probably a recipe?) and link to it using the related recipes component.

This present most recently in #3181 and in the Discord [`#feedback-ideas`](https://discord.com/channels/830184174198718474/872579324446928896/1110884462281097256) channel.

Happy to make a PR once we decide what we want.

### PR DESCRIPTION
**Pull request by TheOtterlord**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Closes #3327
- What does this PR change? Adds a new recipe for sharing state between Astro components. Also adds a link from the "Share State Between Islands" page, and vice-versa.

### CODE REVIEW

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  ```ts title="src/store.js"
+  import { atom } from 'nanostores';
+
+  export const isOpen = atom(false);
+  ```
+
+3. Import and use the store in a [`script`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) tag in your components:
+
+  ```astro title="src/components/Button.astro"
+  <button id="openDialog">Open</button>
```
Comment (dreyfus92) [inline src/content/docs/en/recipes/sharing-state.mdx:49]: What do you think about adding the code fence?
```suggestion
---
---
  <button id="openDialog">Open</button>
```

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    function openDialog() {
+      isOpen.set(true);
+    }
+
+    document.getElementById('openDialog').addEventListener('click', openDialog);
+  </script>
+  ```
+
+  ```astro title="src/components/Dialog.astro"
+  <div id="dialog" style="display: hidden">Hello world!</div>
```
Comment (dreyfus92) [inline src/content/docs/en/recipes/sharing-state.mdx:64]:
```suggestion
---
---
  <div id="dialog" style="display: hidden">Hello world!</div>
```

**Review by dreyfus92 — CHANGES_REQUESTED** [review]
Awesome job @TheOtterlord, I just leave you a couple of small suggestions 😁

**Comment by sarah11918** [comment]
Another Otterlord recipe!  :raised_hands: 

I promise, I'm going to get to these!  :sweat_smile:  Getting some 2.6 launch docs organized first. Thanks for the initial review, @dreyfus92 :rocket:

File: src/content/docs/en/recipes/sharing-state.mdx (hunk 1 above)
Comment (TheOtterlord) [reply src/content/docs/en/recipes/sharing-state.mdx:49]: I removed it originally to reduce scroll-space, but actually it does look a lot more "Astro" with the code fences vs plain hrml. which is nice.

Comment (TheOtterlord) [reply src/content/docs/en/recipes/sharing-state.mdx:51]: added the code fence :D

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  import { atom } from 'nanostores';
+
+  export const isOpen = atom(false);
+  ```
+
+3. Import and use the store in a [`script`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) tag in your components:
+
+  ```astro title="src/components/Button.astro"
+  ---
+  ---
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/sharing-state.mdx:50]: If we don't need the frontmatter, I would opt to totally remove it from the codesample.

```suggestion
```

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      isOpen.set(true);
+    }
+
+    document.getElementById('openDialog').addEventListener('click', openDialog);
+  </script>
+  ```
+
+  ```astro title="src/components/Dialog.astro"
+  ---
+  ---
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/sharing-state.mdx:67]: see comment before
```suggestion
```

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      }
+    })
+
+    document.getElementById('openDialog').addEventListener('click', openDialog);
+  </script>
+  ```
+
+## Resources
+
+- [Nano Stores - NPM](https://www.npmjs.com/package/nanostores)
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/sharing-state.mdx:88]:
```suggestion
- [Nano Stores on NPM](https://www.npmjs.com/package/nanostores)
```

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+type: recipe
+---
+
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+
+:::tip
+  Using Framework components? See [how to share state between Islands](/en/core-concepts/sharing-state/)!
+:::
+
+When building an Astro website, you may need to share state across components. Astro recommend the use of [Nano Stores](https://github.com/nanostores/nanostores) for shared client storage.
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/sharing-state.mdx:14]:
```suggestion
When building an Astro website, you may need to share state across components. Astro recommends the use of [Nano Stores](https://github.com/nanostores/nanostores) for shared client storage.
```

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+2. Create a store. In this example, the store tracks whether a dialog is open or not:
+
+  ```ts title="src/store.js"
+  import { atom } from 'nanostores';
+
+  export const isOpen = atom(false);
+  ```
+
+3. Import and use the store in a [`script`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) tag in your components:
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/sharing-state.mdx:46]: Personal opinion; I don't think that we need to link the HTML docs here. If our readers are trying to implement stores, they probably know what HTML / JS is, thus are familiar with the `script` tag.

What do you think?

```suggestion
3. Import and use the store in a `<script>` tag in your components:
```

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    })
+
+    document.getElementById('openDialog').addEventListener('click', openDialog);
+  </script>
+  ```
+
+## Resources
+
+- [Nano Stores - NPM](https://www.npmjs.com/package/nanostores)
+- [Nano Stores - Vanilla JS docs](https://github.com/nanostores/nanostores#vanilla-js)
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/sharing-state.mdx:89]:
```suggestion
- [Nano Stores documentation for Vanilla JS](https://github.com/nanostores/nanostores#vanilla-js)
```

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+description: Learn how to share state across Astro components — with Nano Stores.
+i18nReady: true
+type: recipe
+---
+
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+
+:::tip
+  Using Framework components? See [how to share state between Islands](/en/core-concepts/sharing-state/)!
+:::
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/sharing-state.mdx:12]:
```suggestion
:::tip
Using Framework components? See [how to share state between Islands](/en/core-concepts/sharing-state/)!
:::
```

**Review by ElianCodes — COMMENTED** [review]
Really like it! Thanks @TheOtterlord!

I just left a few comments / opinions and a small typo!

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+title: Share State Between Astro Components
+description: Learn how to share state across Astro components — with Nano Stores.
+i18nReady: true
+type: recipe
+---
+
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+
+:::tip
+Using Framework components? See [how to share state between Islands](/en/core-concepts/sharing-state/)!
```
Comment (Yan-Thomas) [inline src/content/docs/en/recipes/sharing-state.mdx:11]: Nit: `Framework components` -> `framework components` for consistency with other pages
```suggestion
Using framework components? See [how to share state between Islands](/en/core-concepts/sharing-state/)!
```

**Review by Yan-Thomas — COMMENTED** [review]
Thanks @TheOtterlord, added a small suggestion for you!

File: src/content/docs/en/recipes/sharing-state.mdx (hunk 10 above)
Comment (TheOtterlord) [reply src/content/docs/en/recipes/sharing-state.mdx]: good catch!

File: src/content/docs/en/core-concepts/sharing-state.mdx [hunk 11]
```diff
@@ -1,20 +1,23 @@
 ---
 title: Share State Between Islands
-description: Learn how to share state across components — and frameworks! — with Nano Stores.
+description: Learn how to share state across framework components — with Nano Stores.
```
Comment (sarah11918) [inline src/content/docs/en/core-concepts/sharing-state.mdx:3]:
```suggestion
description: Learn how to share state across framework components with Nano Stores.
```

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 12]
```diff
@@ -0,0 +1,85 @@
+---
+title: Share State Between Astro Components
+description: Learn how to share state across Astro components — with Nano Stores.
```
Comment (sarah11918) [inline src/content/docs/en/recipes/sharing-state.mdx:3]:
```suggestion
description: Learn how to share state across Astro components with Nano Stores.
```

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+  <script>
+    import { isOpen } from '../store.js';
+    
+    // Set the store to true when the button is clicked
+    function openDialog() {
+      isOpen.set(true);
+    }
+
+    document.getElementById('openDialog').addEventListener('click', openDialog);
```
Comment (sarah11918) [inline src/content/docs/en/recipes/sharing-state.mdx:59]:
```suggestion
    // Set the store to true when the button is clicked
    function openDialog() {
      isOpen.set(true);
    }

    // Add an event listener to the button
    document.getElementById('openDialog').addEventListener('click', openDialog);
```

File: src/content/docs/en/recipes/sharing-state.mdx [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+
+2. Create a store. In this example, the store tracks whether a dialog is open or not:
+
+  ```ts title="src/store.js"
+  import { atom } from 'nanostores';
+
+  export const isOpen = atom(false);
+  ```
+
+3. Import and use the store in a `<script>` tag in your components:
```
Comment (sarah11918) [inline src/content/docs/en/recipes/sharing-state.mdx:46]:
```suggestion
3. Import and use the store in a `<script>` tag in the components that will share state.

The following `Button` and `Dialog` components each use the shared `isOpen` state to control whether a particular `<div>` is hidden or displayed:
```

**Review by sarah11918 — COMMENTED** [review]
Hey @TheOtterlord , I really like this one!

I've just added a tiny bit of *context* (not really "explanation") so we can draw attention to the points the reader should focus on.

Otherwise, I think this is great, and when you're happy, go ahead and merge!

**Pull request merged**